pkg/scaling/scale_handler.go

```diff
@@ -388,7 +377,18 @@ func (h *scaleHandler) performGetScalersCache(ctx context.Context, key string, s
 	default:
 	}
 
+	// Scalers Close() could be impacted by timeouts, blocking the mutex
+	// until the timeout happens. Instead of locking the mutex, we take
+	// the old cache item and we close it in another goroutine, not locking
+	// the cache: https://github.com/kedacore/keda/issues/5083
+	if regenerateCache {
+		oldCache := h.scalerCaches[key]
+		go oldCache.Close(ctx)
+	}
+
+	h.scalerCachesLock.Lock()
 	h.scalerCaches[key] = newCache
+	h.scalerCachesLock.Unlock()
 	return h.scalerCaches[key], nil
```

Comment:
```suggestion
    defer h.scalerCachesLock.Unlock()
	h.scalerCachesLock.Lock()
	h.scalerCaches[key] = newCache

	return h.scalerCaches[key], nil
```

Reply:
the original code is correct also, though this is more go idiomatic and safer for any potential future additions

Reply:
Yeah,7d2b612dfc9f971731b30d8ef4f5aa6efbd4e708 and I just removed the `defer` because it's just one line xD

![image](https://github.com/kedacore/keda/assets/36899226/b74b80c0-8c8d-4d48-8364-7a25ef66f8c9)

Reply:
done!